src/libLookup/Lookup.cpp

```diff
@@ -1460,106 +1467,114 @@ bool Lookup::AddMicroBlockToStorage(const MicroBlock& microblock) {
     [[gnu::unused]] const Peer& from) {
-  LOG_GENERAL(WARNING, "Function not in used");
-  return false;
+  LOG_MARKER();
 
-  // LOG_MARKER();
+  if (!LOOKUP_NODE_MODE) {
+    LOG_GENERAL(WARNING,
+                "Function not expected to be called from non-lookup node");
+    return false;
+  }
 
-  // if (!LOOKUP_NODE_MODE) {
-  //   LOG_GENERAL(WARNING,
-  //               "Function not expected to be called from non-lookup node");
-  //   return false;
-  // }
-  // vector<BlockHash> microBlockHashes;
-  // uint32_t portNo = 0;
+  vector<BlockHash> microBlockHashes;
+  uint32_t portNo = 0;
 
-  // if (!Messenger::GetLookupGetMicroBlockFromLookup(message, offset,
-  //                                                  microBlockHashes, portNo))
-  //                                                  {
-  //   LOG_GENERAL(WARNING, "Failed to process");
-  //   return false;
-  // }
+  if (!Messenger::GetLookupGetMicroBlockFromLookup(message, offset,
+                                                   microBlockHashes, portNo)) {
+    LOG_GENERAL(WARNING, "Failed to process");
+    return false;
+  }
 
-  // if (microBlockHashes.size() == 0) {
-  //   LOG_GENERAL(INFO, "No MicroBlock requested");
-  //   return true;
-  // }
+  if (microBlockHashes.size() == 0) {
+    LOG_GENERAL(INFO, "No MicroBlock requested");
+    return true;
+  }
 
-  // LOG_GENERAL(INFO, "Reques for " << microBlockHashes.size() << " blocks");
-
-  // uint128_t ipAddr = from.m_ipAddress;
-  // Peer requestingNode(ipAddr, portNo);
-  // vector<MicroBlock> retMicroBlocks;
-
-  // for (const auto& mbhash : microBlockHashes) {
-  //   LOG_GENERAL(INFO, "[SendMB]"
-  //                         << "Request for microBlockHash " << mbhash);
-  //   shared_ptr<MicroBlock> mbptr;
-  //   if (!BlockStorage::GetBlockStorage().GetMicroBlock(mbhash, mbptr)) {
-  //     LOG_GENERAL(WARNING, "Failed to fetch micro block Hash " << mbhash);
-  //     continue;
-  //   } else {
-  //     retMicroBlocks.push_back(*mbptr);
-  //   }
-  // }
+  LOG_GENERAL(INFO, "Reques for " << microBlockHashes.size() << " blocks");
 
-  // bytes retMsg = {MessageType::LOOKUP,
-  //                 LookupInstructionType::SETMICROBLOCKFROMLOOKUP};
+  uint128_t ipAddr = from.m_ipAddress;
+  Peer requestingNode(ipAddr, portNo);
+  vector<MicroBlock> retMicroBlocks;
 
-  // if (retMicroBlocks.size() == 0) {
-  //   LOG_GENERAL(WARNING, "return size 0 for microblocks");
-  //   return true;
-  // }
+  for (const auto& mbhash : microBlockHashes) {
+    LOG_GENERAL(INFO, "[SendMB]"
+                          << "Request for microBlockHash " << mbhash);
+    shared_ptr<MicroBlock> mbptr;
+    int retryCount = 5;
 
-  // if (!Messenger::SetLookupSetMicroBlockFromLookup(
-  //         retMsg, MessageOffset::BODY, m_mediator.m_selfKey, retMicroBlocks))
-  //         {
-  //   LOG_GENERAL(WARNING, "Failed to Process ");
-  //   return false;
-  // }
+    while (retryCount-- > 0) {
```

Comment:
Do we really need retry on lookup while responding to request for mbs?

Reply:
Yes we need it because it's a kind of late-request, and in worst case some previous mb even will be received after the later mb.